src/Raven.Server/Documents/TimeSeries/TimeSeriesValuesSegment.cs

```diff
@@ -468,7 +473,13 @@ public IEnumerable<SingleResult> YieldAllValues(JsonOperationContext context, By
                     if (status == Dead && includeDead == false)
                         continue;
 
-                    var cur = baseline.AddMilliseconds(ts);
+                    var next = baseline.AddMilliseconds(ts);
+                    if (next == current && 
+                        Version == SegmentVersion.V50000) // fix legacy issue RavenDB-15617
```

Comment:
What is the difference between the versions?

Reply:
different issue...  we might have duplicate timestamps :/